src/java.base/unix/classes/java/lang/ProcessImpl.java

```diff
@@ -428,7 +401,8 @@ void initStreams(int[] fds, boolean forceNullOutputStream) throws IOException {
                 });
                 break;
 
-            default: throw new AssertionError("Unsupported platform: " + platform);
+            case Windows: throw new AssertionError("Unsupported platform: " +
+                    OperatingSystem.current());
```

Comment:
This is switch on a enum so if a new port is added then this will just fall through silently, whereas I think you want it to fail.